src/index.ts

```diff
@@ -485,7 +485,7 @@ export class AzureOpenAI extends OpenAI {
   }
 
   protected override async prepareOptions(opts: Core.FinalRequestOptions<unknown>): Promise<void> {
-    if (opts.headers?.['Authorization'] || opts.headers?.['api-key']) {
```

Comment:
Will this will overwrite an Authorization header passed by the user?

Reply:
We don't support the scenario of passing a single AAD token in the authorization header manually. The client constructor requires the user to provide either an API key or an AAD token provider.